### ISSUE TITLE
Simulate voltage-clamp protocols faster

### ISSUE THREAD

**Issue by MichaelClerx** [ModellingWebLab/project_issues#20]
- ~Test 'method 2' for HH models in Myokit, finish mini-report~
- ~Auto-detect HH models, use explicit solution?~
- ~Or rush-larsen?~
- ~Auto-detect Markov models, use explicit solution?~

Update (2018-11-02):

Eventually, we should have some machinery in the WL for running clamp-only protocols on Markov or HH models, both of which can be solved analytically for fixed voltages.

**Comment by mirams** [comment]
Yep - just having a nice tidy solver that does exact solutions for MM and HH models for flat voltage steps, then swaps to CVODE for sine wave section, should give nice speed ups.

I would like to get these kind of solvers auto-generated from CellML. Would require figuring out what the transition matrix is for Markov models specified as a system of ODEs.

**Comment by MichaelClerx** [comment]
> I would like to get these kind of solvers auto-generated from CellML. Would require figuring out what the transition matrix is for Markov models specified as a system of ODEs.

Can copy/paste that from Myokit when the time is right! (It uses sympy for this step)

**Comment by MichaelClerx** [comment]
I'd say this is out of scope now, or possibly a weblab_cg nice-to-have?

Can I close this @jonc125 @mirams ?

**Comment by mirams** [comment]
Can we have a WL3 tag for longer future tickets, rather than closing them and never finding them again?

**Comment by jonc125** [comment]
We certainly don't have to close issues that are out of scope for the BBSRC grant, since everything *in* scope should be linked to an epic, and we can use those to filter tasks of interest.
